MIPs/mip-mc-request.md

```diff
@@ -0,0 +1,156 @@
+---
+MIP: X
+Title: Multichain Provider Request Routing
+Status: Draft
+Stability: n/a
+discussions-to: https://github.com/MetaMask/metamask-improvement-proposals/discussions
+Author(s): Vandan Parikh (@vandan), Alex Donesky (@adonesky1)
+Type: Maintainer
+Created: 2024-01-25
+---
+
+## Summary
+The proposal aims to specify the implementation of the [CAIP-27](https://github.com/ChainAgnostic/CAIPs/blob/main/CAIPs/caip-27.md) standard for multi-chain wallet API calls in MetaMask, providing a clear and standardized way for dApps to make JSON-RPC requests that can be scoped to a specific chain. By wrapping typical JSON-RPC calls within a "provider_request" that includes a "scope," developers can specify which chain the request is intended for.
+
+## Motivation
+Both users and dapp builders are increasingly interested in experiences that involve multiple chains. MetaMask shouldn't get in their way.
+
+This proposal is intended to support the following objectives:
+- Unlocking user experiences that involve minimal friction when interacting across different chains (both in dApps and MetaMask).
+- Simplifying development for dapp builders.
+- Aligning MetaMask with community standards to simplify integrations and tooling in the ecosystem.
+- Increasing MetaMask's utility as a universal wallet solution.
+
+Improved conversion rates on current activities that involve multiple chains can be realized through the Multichain API. More importantly, the implementation of a multichain API will serve as a significant enabler of permissionless innovation in the community. We anticipate that it will drive dramatic growth in existing and entirely new categories of dApps.
+
+# Proposal
+
+## Language
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" written in uppercase in this document are to be interpreted as described in RFC 2119.
+
+## Definitions
+
+Add a note about how we use the terms `chain` and `network` interchangeably
```

Comment:
haha this is an "ethereumism" and some non-chain networks like DAGs might feel more welcome in the Snaps project if you discriminate!